tools/generate-zipapp.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+BUILDDIR=$(mktemp -d -q)
+cleanup() {
+    rm -rf "$BUILDDIR"
+}
+trap cleanup EXIT
+
+mkdir -p builddir
+
+cp -r mkosi "${BUILDDIR}/"
+
+python3 -m zipapp \
+        -p "/usr/bin/env python3" \
+        -o builddir/mkosi \
```

Comment:
I think I'm missing something, won't this delete the mkosi binary immediately after we build it with the trap?

I would copy mkosi dir to builddir, build the zipapp as mkosi.pyz (or whatever the default extension is), delete $BUILDDIR/mkosi and rename mkosi.pyz to mkosi and keep it in the builddir.